ngraph/frontend/paddlepaddle/src/op/pool2d.cpp

```diff
@@ -25,47 +15,227 @@ namespace ngraph
+                            pool_size_Height = pool_size_Width = kernel_shape[0];
+                        }
+                        else
+                        {
+                            pool_size_Height = kernel_shape[0];
+                            pool_size_Width = kernel_shape[1];
+                        }
+
+                        uint64_t stride_h = int64_t(input_h / pool_size_Height);
+                        uint64_t stride_w = int64_t(input_w / pool_size_Width);
+                        uint64_t kernel_h = input_h - (pool_size_Height - 1) * stride_h;
+                        uint64_t kernel_w = input_w - (pool_size_Width - 1) * stride_w;
+
+                        if (stride_h < 1 || stride_w < 1)
+                        { // upsampling?
+                            throw std::runtime_error("Unsupported pooling adaptive type!");
+                        }
+
+                        if (pooling_type == "max")
+                        {
+                            std::cout << "Convert MAx" << std::endl;
+                            return node.default_single_output_mapping(
+                                {std::make_shared<ngraph::opset6::MaxPool>(
+                                    data,
+                                    ngraph::Strides{stride_h, stride_w},
+                                    pad_begin,
+                                    pad_end,
+                                    ngraph::Shape{kernel_h, kernel_w},
+                                    rounding_type,
+                                    auto_pad)},
+                                {"Out"});
+                        }
+                        else
+                        {
+                            bool exclude_pad = node.get_attribute<bool>("exclusive") ? true : false;
+                            return node.default_single_output_mapping(
+                                {std::make_shared<ngraph::opset6::AvgPool>(
+                                    data,
+                                    ngraph::Strides{stride_h, stride_w},
+                                    pad_begin,
+                                    pad_end,
+                                    ngraph::Shape{kernel_h, kernel_w},
+                                    exclude_pad,
+                                    rounding_type,
+                                    auto_pad)},
+                                {"Out"});
+                        }
                     }
                     else
                     {
-                        throw std::runtime_error("Unsupported pooling type");
+                        auto strides = node.get_attribute<std::vector<int32_t>>("strides");
+                        auto paddings = node.get_attribute<std::vector<int32_t>>("paddings");
+
+                        uint64_t kernel_h, kernel_w;
+                        if (kernel_shape.size() == 1)
+                        {
+                            kernel_h = kernel_w = kernel_shape[0];
```

Comment:
This case is not covered by unit test. If it is not possible to generate model with current framework, I'd suggest add a note here about that

Reply:
@ceciliapeng2011, could you please fix this one ?